src/main/java/com/sap/oss/phosphor/fosstars/data/github/experimental/SecurityReviewsFromOpenSSF.java

```diff
@@ -0,0 +1,209 @@
+  }
+
+  /**
+   * Reads a review date from metadata.
+   *
+   * @param metadata The metadata.
+   * @return A review date if available.
+   */
+  Optional<Date> readReviewDateFrom(JsonNode metadata) {
+    JsonNode node = metadata.at("/Review-Date");
+    if (!node.isTextual()) {
+      logger.warn("Oops! Review date is not a string!");
+      return Optional.empty();
+    }
+
+    try {
+      return Optional.of(DATE_FORMAT.parse(node.asText()));
+    } catch (ParseException e) {
+      logger.warn(() -> String.format("Oops! Could not parse date: %s", node.asText()), e);
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Checks if a review was done for a project.
+   *
+   * @param project The project.
+   * @param metadata Metadata of the review.
+   * @return True if the review was done for the project, false otherwise.
+   */
+  boolean isReviewFor(GitHubProject project, JsonNode metadata) {
+    JsonNode node = metadata.at("/Package-URLs");
+    if (!node.isArray()) {
+      logger.warn("Oops! Package-URLs is not an array!");
+      return false;
+    }
+
+    return StreamSupport.stream(node.spliterator(), false)
+        .filter(JsonNode::isTextual)
+        .map(JsonNode::asText)
+        .anyMatch(purl -> purlBelongsTo(project, purl));
+  }
+
+  /**
+   * Checks if a PURL belongs to a project.
+   *
+   * @param project The project.
+   * @param purl The PURL.
+   * @return True if the PURL belongs to the project, false otherwise.
+   */
+  boolean purlBelongsTo(GitHubProject project, String purl) {
+    try {
+      PackageURL packageUrl = new PackageURL(purl);
+      return "github".equalsIgnoreCase(packageUrl.getType())
+          && packageUrl.getNamespace() != null
+          && packageUrl.getName() != null
+          && project.equals(new GitHubProject(packageUrl.getNamespace(), packageUrl.getName()));
+    } catch (MalformedPackageURLException e) {
+      logger.warn(() -> format("Oops! Could not parse package URL: %s", purl), e);
+    }
+
+    return false;
+  }
+
+  /**
+   * Reads review metadata from a file.
+   *
+   * @param file The file.
+   * @return Metadata if available.
+   * @throws IOException If something went wrong.
+   */
+  private static Optional<JsonNode> readMetadataFrom(Path file) throws IOException {
+    try (BufferedReader reader = Files.newBufferedReader(file)) {
+      return readMetadataFrom(reader);
+    }
+  }
+
+  /**
+   * Reads review metadata from a reader.
+   *
+   * @param reader The reader.
+   * @return Metadata if available.
+   * @throws IOException If something wrong.
+   */
+  static Optional<JsonNode> readMetadataFrom(BufferedReader reader) throws IOException {
+    String line = reader.readLine();
+    if (!"---".equals(line)) {
+      return Optional.empty();
+    }
+
+    StringBuilder metadata = new StringBuilder();
+    do {
+      metadata.append(line).append("\n");
+      line = reader.readLine();
+    } while (line != null && !"---".equals(line));
```

Comment:
Shouldn't this be `line != null || !"---".equals(line)`

Reply:
Nope, it has to stop when the line is `---`.